src/catalog/src/durable/impls/persist.rs

```diff
@@ -158,20 +200,56 @@ impl UnopenedPersistCatalogState {
             )
             .await
             .expect("invalid usage");
-        let current_snapshot =
-            current_snapshot_binary(&mut write_handle, &mut read_handle, &metrics).await;
-        let epoch = current_snapshot
-            .as_ref()
-            .map(|(snapshot, _)| extract_epoch(snapshot.iter()));
+
+        // Commit an empty write at the minimum timestamp so the catalog is always readable.
+        const EMPTY_UPDATES: &[((SourceData, ()), Timestamp, Diff)] = &[];
+        let _ = write_handle
+            .compare_and_append(
+                EMPTY_UPDATES,
+                Antichain::from_elem(Timestamp::minimum()),
+                Antichain::from_elem(Timestamp::minimum().step_forward()),
+            )
+            .await
+            .expect("invalid usage");
```

Comment:
I'm not in love with this because someone might be connecting with the intention of only performing reads. From that read-only context I'd prefer not to attempt any writes at all. Unfortunately, at this point we don't know yet whether they will perform reads or writes. I couldn't come up with a better solution though.

@danhhz if someone tries executing this with a read-only CRDB user, will they get an `InvalidUsage` error, an `UpperMismatch` error, or will it hang forever?

Reply:
> @danhhz if someone tries executing this with a read-only CRDB user, will they get an InvalidUsage error, an UpperMismatch error, or will it hang forever?

I just tested locally and it hangs forever.

Reply:
Not surprised! I also wouldn't particularly consider this "out of spec" for persist. My guess is it would start logging the errors at info after a while?

Reply:
It actually never even makes it here and hangs while trying to open a critical since handle:

```
2024-01-19T17:35:27.864629Z  INFO new{organization_id=0b1dcfa8-1239-4722-9101-b1c1577264fb}:open_critical_since{shard=sf57e434d-494e-a042-da45-73763d2d3706}: mz_persist_client::internal::machine: apply_unbatched_idempotent_cmd register received an indeterminate error, retrying in 8.192s: indeterminate: db error: ERROR: user read_only does not have INSERT privilege on relation consensus
```

So I'm actually not too worried.